### PR TITLE
Feat/[profiles]/db technologies filter

### PR DESCRIPTION
**Pull request by szymonkadas**
# Summary: 
- technology filter is now showing all technologies used by user in db.
- filter shows 8 rows of data (I recommend checking on scss file)

# Notes
- didn't apply any changes to phone display
- as i can see on figma, filters layout is currently kinda off, could try to fix it if needed (would work on phone display aswell)

### CODE REVIEW

File: src/data/frontend/profile/mappers.ts [hunk 1]
```diff
@@ -82,15 +84,11 @@ export const mappedLocations = ['Poland', 'Europe', 'Other'].map(
...
-  'Node.js',
-  'React.js',
-  'Vue.js',
-  'Angular',
-  'MongoDB',
-].map((technology) => ({
-  name: technology,
-  value: technology,
-}))
+export const mappedTechnologies: Promise<DropdownOption[]> = apiClient
```
Comment (Sumick) [inline src/data/frontend/profile/mappers.ts:87]: **issue:** mappers are only responsible for mapping data from one type to other type / format - not fetching it

File: src/app/api/filters/technologies/route.ts
Comment (Sumick) [inline src/app/api/filters/technologies/route.ts:1]: **nit:** we don't need this endpoint as we can use React server component to fetch that data

File: src/backend/filters/filters.service.ts
Comment (Sumick) [inline src/backend/filters/filters.service.ts:1]: **nit:** File name should be the same as domain, so `backend/technology/technology.service.ts`

File: src/contexts/FilterContext.tsx [hunk 2]
```diff
@@ -95,7 +116,20 @@ export const FiltersProvider = ({
   const [availabilityFilter, setAvailabilityFilter] = useState<FilterOption[]>(
     [],
   )
-
+  const [filterLists, setFilterLists] = useState<FilterLists>({
+    technology: [],
+    seniority: mappedSeniorityLevel,
```
Comment (Sumick) [inline src/contexts/FilterContext.tsx:121]: **nit:** We can remove this state as we will do it using RSC

File: src/contexts/FilterContext.tsx [hunk 3]
```diff
@@ -95,7 +116,20 @@ export const FiltersProvider = ({
...
+    availability: mappedEmploymentType,
+    location: mappedLocations,
+  })
+  useEffect(() => {
+    mappedTechnologies.then((technologies) =>
+      setFilterLists((prev: FilterLists) => ({
```
Comment (Sumick) [inline src/contexts/FilterContext.tsx:127]: **nit:** we can also remove that useEffect

File: src/lib/apiClient.ts [hunk 4]
```diff
@@ -81,4 +81,15 @@ export const apiClient = {
       throw new Error(error as string)
     }
   },
+
+  getTechnologies: async () => {
```
Comment (Sumick) [inline src/lib/apiClient.ts:85]: **nit:** we won't need this endpoint, as we will use RSC

**Review by Sumick — CHANGES_REQUESTED** [review]
There is a lot of stuff that we can remove because of using react server component. Just do following, so you can remove all other stuff:


```ts
interface FiltersProps {
   technologies: Technology[]
}

const Filters: React.FC = (props: FiltersProps) => {
  ...
}
```

```ts
const FiltersWithData = async () => {
const technologies = await getTechnologies()
return  <Filters technologies={technologies} />
}
```
```ts
export default function Home() {
  return (
    <FiltersProvider>
      <FiltersWithData />
      <ProfileList />
    </FiltersProvider>
  )
}
```

**Review by Sumick — APPROVED** [review]
Very nice:)

**Review by wawrzonnn — APPROVED** [review]

**Review by hansac91 — APPROVED** [review]

**Pull request merged**